integration-tests/src/test/java/io/apicurio/tests/ConfluentBaseIT.java

```diff
@@ -68,16 +68,15 @@ public int createArtifactViaConfluentClient(ParsedSchema schema, String artifact
     }
 
     protected void clearAllConfluentSubjects() throws IOException, RestClientException {
-        for (String confluentSubject : confluentService.getAllSubjects()) {
-            logger.info("Deleting confluent schema {}", confluentSubject);
-            try {
-                confluentService.deleteSubject(confluentSubject);
-            } catch (RestClientException e) {
-                if (e.getStatus() == 404) {
-                    //subjects may be already deleted
-                    continue;
+        if (confluentService != null && !confluentService.getAllSubjects().isEmpty()) {
+            for (String confluentSubject : confluentService.getAllSubjects()) {
+                logger.info("Deleting confluent schema {}", confluentSubject);
+                try {
+                    registryClient.deleteArtifact("default", confluentSubject);
+                } catch (Exception e) {
+                    System.out.println("WARNING>> Failed to delete confluent schema " + confluentSubject);
```

Comment:
Are there other tests that rely on the cleanup? I think we should keep the original check, and either fail the test or recursively delete the referring schemas. In the first case the test author would have to cleanup, the second case would be more difficult to implement and depend on finding out the dependencies from the error message.

Reply:
Multiple tests rely on the cleanup and fail if it doesn't work.  Before making this change, I tried making multiple passes (using a while loop instead of an if block) - so it would keep iterating over all the subjects until it couldn't find any more, but that didn't work and I didn't spend time figuring out why.  :(

Making the tests responsible for their own cleanup is sensible.  I would be happy with such an improvement, although this change represents an "equivalent" approach to what we did before, which I think is sufficient for this PR.